src/probnum/quad/solvers/belief_updates/_belief_update.py

```diff
@@ -74,47 +132,54 @@ def __call__(
+                    new_kernel_embedding.kernel_mean(new_nodes),
                 )
             )
 
-        initial_integral_variance = bq_state.kernel_embedding.kernel_variance()
-        weights = self._solve_gram(gram, kernel_means)
+        # Cholesky factorisation of the Gram matrix
+        gram_cho_factor = self._compute_gram_cho_factor(gram)
 
-        # integral mean and variance
+        # Estimate scaling parameter
+        new_scale_sq = self._estimate_scale(fun_evals, gram_cho_factor, bq_state)
+
+        # Integral mean and variance
+        weights = cho_solve(gram_cho_factor, kernel_means)
         integral_mean = weights @ fun_evals
-        integral_variance = initial_integral_variance - weights @ kernel_means
+        initial_integral_variance = new_kernel_embedding.kernel_variance()
+        integral_variance = new_scale_sq * (
+            initial_integral_variance - weights @ kernel_means
+        )
 
-        updated_belief = Normal(integral_mean, integral_variance)
-        updated_state = BQState.from_new_data(
+        new_belief = Normal(integral_mean, integral_variance)
+        new_state = BQState.from_new_data(
+            kernel=new_kernel,
+            scale_sq=new_scale_sq,
             nodes=nodes,
             fun_evals=fun_evals,
-            integral_belief=updated_belief,
+            integral_belief=new_belief,
             prev_state=bq_state,
             gram=gram,
             kernel_means=kernel_means,
         )
 
-        return updated_belief, updated_state
-
-    @staticmethod
-    def _solve_gram(gram: np.ndarray, rhs: np.ndarray) -> np.ndarray:
-        """Solve the linear system of the form.
-
-        .. math:: Kx=b,
-
-        Parameters
-        ----------
-        gram :
-            symmetric pos. def. kernel Gram matrix :math:`K`, shape (nevals, nevals)
-        rhs :
-            right-hand-side :math:`b`, matrix or vector, shape (nevals, ...)
-
-        Returns
-        -------
-        x :
-            The solution to the linear system :math:`K x = b`
-        """
-        jitter = 1.0e-6
-        chol_gram = cho_factor(gram + jitter * np.eye(gram.shape[0]))
-        return cho_solve(chol_gram, rhs)
+        return new_belief, new_state
+
+    # pylint: disable=no-self-use
+    def _estimate_kernel(self, kernel: Kernel) -> Tuple[Kernel, bool]:
+        """Estimate the intrinsic kernel parameters. That is, all parameters except the
+        scale."""
+        new_kernel = kernel
+        kernel_was_updated = False
+        return new_kernel, kernel_was_updated
+
+    def _estimate_scale(
+        self, fun_evals: np.ndarray, gram_cho_factor: np.ndarray, bq_state: BQState
+    ) -> FloatLike:
+        """Estimate the scale parameter."""
+        if self.scale_estimation == "mle":
+            new_scale_sq = (
+                fun_evals @ cho_solve(gram_cho_factor, fun_evals) / fun_evals.shape[0]
+            )
+        else:
```

Comment:
The `else` statement captures everything apart from "mle", but it should only capture `None` and raise an exception otherwise.